src/test/utils/singleton_test.cpp

```diff
@@ -0,0 +1,50 @@
+#include "../base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "storage/storage_manager.hpp"
+#include "utils/plugin_manager.hpp"
+#include "utils/singleton.hpp"
+
+#include "./plugin_test_utils.hpp"
+#include "./test_plugin.hpp"
+
+namespace opossum {
+
+class SingletonTest : public BaseTest {
+ protected:
+  std::unordered_map<PluginName, PluginHandleWrapper>& get_plugins() {
+    auto& pm = PluginManager::get();
+
+    return pm._plugins;
+  }
+};
+
+TEST_F(SingletonTest, SingleInstance) {
+  auto& a = Singleton<int>::get();
+  auto& b = Singleton<int>::get();
+
+  EXPECT_EQ(&a, &b);
+}
+
+// ASAN cannot handle the not yet defined (because it's a dynamic library) typeinfo for TestPlugin.
+// Therefore, this test is only built if ASAN is not activated.
+#if defined(__has_feature)
```

Comment:
Don't we have a blacklist for asan, ".asan_ignore", something? Seems invasive to alter code whenever CI tools run into problems

Reply:
asan_ignore is used during actual runtime of the binary built with address sanitisation. The check I introduced is applied during compilation, without it, the tests wouldn't compile with address sanitisation enabled.